flink-filesystems/flink-gs-fs-hadoop/src/main/java/org/apache/flink/fs/gs/GSFileSystemOptions.java

```diff
@@ -80,6 +84,74 @@ public class GSFileSystemOptions {
                     .withDescription(
                             "This option sets the timeout in milliseconds to read data from an established connection.");
 
+    /**
+     * Flink config option to set the http read timeout. It will be used by cloud-storage library.
+     */
+    public static final ConfigOption<Integer> GCS_RETRY_MAX_ATTEMPT =
+            ConfigOptions.key("gs.retry.max-attempt")
+                    .intType()
+                    .defaultValue(6)
```

Comment:
I'm not sure about setting default values for these options. The default values are ignored anyway because these options are only used in `getOptional()`, which returns `empty()` if the option is not explicitly configured by user. We can mention in the description that if not configured, the GCS default values will be used.